samples/nrf9160/modem_shell/src/ppp/ppp_ctrl.c

```diff
@@ -52,16 +53,17 @@ uint16_t used_mtu_mru;
 /* Work for auto starting/stopping ppp according to default PDN
  * activation status
  */
-struct ppp_ctrl_worker_data {
+struct ppp_ctrl_pdn_status_worker_data {
 	struct k_work work;
 	bool default_pdn_active;
-} ppp_ctrl_worker_data;
+} ppp_ctrl_pdn_status_worker_data;
 
 static const struct device *ppp_uart_dev;
 
-/* Work for a starting/stopping net if */
-static struct k_work ppp_net_if_start_work;
-static struct k_work ppp_net_if_stop_work;
+/* Work for a re-starting PPP */
+static struct k_work ppp_ctrl_restart_work;
+
+static bool ppp_closing;
```

Comment:
Would it make sense to make this `volatile`, because it's accessed from different threads, I guess?

Reply:
Changed.